planetary-social/nos
# Add a New notes available notification to the Feed

## Issues covered
#1070

## Description
Shows a "New notes available" button when there are new notes to display. Tapping the button reloads the Feed.

In order to match the design, I updated the secondary action gradient colors, so that changes a few other buttons in the app, like:
- the Edit Profile button
- the circular Unfollow (checkmark) button
- the Save button on the Relays screen
- the purple buttons on the Settings screen
- the Warning view button to "View this post anyway"
- the Tap to refresh button on the Profile view

This sounded to me like what Linda wanted in her comment on #1070. And I wanted consistency in the app, so hopefully we're closer on that.

I also did a little or maybe a lot of refactoring to the `PagedNoteListView` and `PagedNoteDataSource`. You'll see -- I think this simplifies things a bit.

Matt and I discussed the `RefreshController` in a call, and for now it's very simple. In the future, we could refactor more functionality into it if we decide that makes sense. It might.

## How to test
1. Navigate to Feed
2. Tap Post
3. Post a note
4. Observe "New notes available" button
5. Tap it
6. Observe the refreshing of the Feed

You can also test by using two instances of the app:

1. Run from this branch and log in with an account that follows one of your test accounts.
2. In another instance of the app, log in with that test account.
3. Post a note from the test account.
4. Observe in the build from this branch that the "New notes available" button appears soon after posting with the other account.

## Screenshots/Video
https://github.com/user-attachments/assets/5f44e314-ce16-4f30-adbe-b9e1ec427aee
